libs/utils/android/workloads/pcmark.py

```diff
@@ -0,0 +1,176 @@
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+import time
+import json
+
+from subprocess import Popen, PIPE
+from zipfile import ZipFile
+
+from android import Screen, System
+from android.workload import Workload
+
+PCMARK_TESTS = ['work']
+
+INPUT_RESULT_FILE  = 'Result.xml'
+OUTPUT_RESULT_FILE = 'scores.json'
+
+# Regexps for benchmark synchronization
+REGEXPS = {
+    'start'  : '.*START.*com.futuremark.pcmark.android.benchmark',
+    'end'    : '.*onWebViewReady.*view_scoredetails.html',
+    'result' : '.*received result for correct code, result file in (?P<path>.*\.zip)',
+    'score'  : '\s*<result_Pcma(?P<name>.*)Score>(?P<score>[0-9]*)<'
+}
+
+class PCMark(Workload):
+    """
+    Android PCMark workload
+
+    APK origin: http://www.futuremark.com/downloads/pcmark-android.apk
+
+    Version 2.0.3716 is known to work
+    """
+
+    # Package required by this workload
+    package  = 'com.futuremark.pcmark.android.benchmark'
+    # Launch activity
+    activity = 'com.futuremark.gypsum.activity.SplashPageActivity'
+
+    def __init__(self, test_env):
+        super(PCMark, self).__init__(test_env)
+        self._log = logging.getLogger('PCMark')
+
+    def run(self, out_dir, test_name, collect=''):
+        """
+        Run single PCMark workload. Returns a collection of results.
+
+        :param out_dir: Path to experiment directory on the host
+                        where to store results.
+        :type out_dir: str
+
+        :param test_name: Name of the test to run
+        :type test_name: str
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+            - 'systrace'
+            - 'ftrace'
+            - any combination of the above as a single space-separated string.
+        :type collect: list(str)
+        """
+
+        self.out_dir = out_dir
+        self.collect = collect
+
+        # Unlock device screen (assume no password required)
+        System.menu(self._target)
+        # Make sure we exit the app if already open
+        System.back(self._target)
+
+        # Close and NOT clear application (benchmark tests are downloaded from web)
+        System.force_stop(self._target, self.package, clear=False)
+
+        # Set airplane mode
+        System.set_airplane_mode(self._target, on=True)
+
+        # Set min brightness
+        Screen.set_brightness(self._target, auto=False, percent=0)
+
+        # Force screen in PORTRAIT mode
+        Screen.set_orientation(self._target, portrait=True)
+
+        monitor = self._target.get_logcat_monitor(REGEXPS.values())
+        
+        # Start PCMark on the target device
+        System.start_activity(self._target, self.package, self.activity)
+        # Wait a few seconds while application is being loaded
+        time.sleep(5)
+        
+        if test_name.lower() == 'work':
+            # Move to benchmark run page
+            self._target.execute('input keyevent KEYCODE_TAB')
```

Comment:
I just remembered that I added these bits in System back in the day of https://github.com/ARM-software/lisa/pull/354/commits/65402030cffe1dde4513b34ea9adf290c2f047d2, but it might be best not pile more stuff up on that class.

Reply:
Do you want me to grab that patch and add it to this PR ? It used to live in the Corporate Geekbench PR but it has been closed since there is support in wltest.